src/data/transform/mod.rs

```diff
@@ -411,21 +429,117 @@ mod tests {
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'true', expect: true",
+                raw_input: String::from("true"),
+                expect_failure: false,
+                expected_value: Some(json!(true)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'true', type: Boolean, expect: true",
+                raw_input: String::from("true"),
+                expect_failure: false,
+                expected_value: Some(json!(true)),
+                column_type: Some(ColumnType::Boolean),
+            },
+            TestCase {
+                name: "converting: 'false', expect: false",
+                raw_input: String::from("false"),
+                expect_failure: false,
+                expected_value: Some(json!(false)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'false', type: Boolean, expect: false",
+                raw_input: String::from("false"),
+                expect_failure: false,
+                expected_value: Some(json!(false)),
+                column_type: Some(ColumnType::Boolean),
+            },
+            TestCase {
+                name: "converting: '42.42', expect: 42.42",
+                raw_input: String::from("42.42"),
+                expect_failure: false,
+                expected_value: Some(json!(42.42)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'my string', expect: 'my string'",
+                raw_input: String::from("my string"),
+                expect_failure: false,
+                expected_value: Some(json!("my string")),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'my string', type: String, expect: 'my string'",
+                raw_input: String::from("my string"),
+                expect_failure: false,
+                expected_value: Some(json!("my string")),
+                column_type: Some(ColumnType::String),
+            },
+            TestCase {
+                name: "converting: 'my string', type: Number, expect: Failure",
+                raw_input: String::from("my string"),
+                expect_failure: true,
+                expected_value: None,
+                column_type: Some(ColumnType::Number),
+            },
+            TestCase {
+                name: "converting: 'my string', type: Boolean, expect: Failure",
+                raw_input: String::from("my string"),
+                expect_failure: true,
+                expected_value: None,
+                column_type: Some(ColumnType::Boolean),
+            },
+        ];
+
+        for test_case in test_cases.iter() {
+            let value =
+                get_json_value_from_string(test_case.raw_input.as_str(), &test_case.column_type);
```

Comment:
In general the prefered way to get a "string slice" from an owned "string" ist just borrowing it with a `&` prefix
```suggestion
                get_json_value_from_string(&test_case.raw_input, &test_case.column_type);
```